### PR TITLE
Rework parameter test to not using launch testing

### PR DESCRIPTION
**Pull request by griswaldbrooks**
## Description

Reworks the parameter test to not use launch_testing

## Testing

```
colcon build
colcon test
```

## Notes

Please add any notes to help reviewers understand how this code should be reviewed.

Be sure to not only [request reviewers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) but also [assign someone to the pull request](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) so it is clear when a review is required.

### CODE REVIEW

File: functional_programming_tests/test/with_parameters.cpp [hunk 1]
```diff
@@ -476,6 +484,76 @@ TEST_F(TaskPlanningFixture, path_generated) {
...
 
+TEST_F(TaskPlanningFixture, large_footprint_no_path) {
+  // Creare executor to spin the node in a separate thread
+  auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+
+  // GIVEN a PathGenerator node with a large robot footprint
+  auto options = rclcpp::NodeOptions{};
+  options.parameter_overrides(
+      std::vector<rclcpp::Parameter>{{"robot_size", 5}});
+  auto const pg = std::make_shared<PathGenerator>(options);
```
Comment (bgill92) [inline functional_programming_tests/test/with_parameters.cpp:495]: Whoa, thats neat

**Review by bgill92 — APPROVED** [review]
LGTM :+1:

File: functional_programming_tests/test/with_parameters.cpp (hunk 1 above)
Comment (griswaldbrooks) [reply functional_programming_tests/test/with_parameters.cpp:495]: What's neat?

Comment (bgill92) [reply functional_programming_tests/test/with_parameters.cpp:495]: Parameter overriding via `NodeOptions`

**Pull request merged**